common/src/cams/utilities.test.ts

```diff
@@ -0,0 +1,93 @@
+import { keyValuesToMap, keyValuesToRecord } from './utilities';
+
+describe('Key Value utils', () => {
+  describe('keyValuesToRecord', () => {
+    test('should turn a key value string into a record', () => {
+      const encoded = 'ONE=1|TWO=2';
+      const expected = {
+        ONE: '1',
+        TWO: '2',
+      };
+      const actual = keyValuesToRecord(encoded);
+      expect(actual).toEqual(expected);
+    });
+
+    test('should turn a blank string into an empty record', () => {
+      const encoded = '';
+      const expected = {};
+      const actual = keyValuesToRecord(encoded);
+      expect(actual).toEqual(expected);
+    });
+
+    test('should omit a pair without a equal sign delimiter', () => {
+      const encoded = 'foo|ONE=1|bad';
+      const expected = {
+        ONE: '1',
+      };
+      const actual = keyValuesToRecord(encoded);
+      expect(actual).toEqual(expected);
+    });
+
+    test('should omit empty pairs', () => {
+      const encoded = '||ONE=1|||||';
+      const expected = {
+        ONE: '1',
+      };
+      const actual = keyValuesToRecord(encoded);
+      expect(actual).toEqual(expected);
+    });
+
+    test('should trim keys and values', () => {
+      const encoded = ' ONE = 1 | TWO      = 2         ';
+      const expected = {
+        ONE: '1',
+        TWO: '2',
+      };
+      const actual = keyValuesToRecord(encoded);
+      expect(actual).toEqual(expected);
+    });
+  });
+
+  describe('keyValuesToMap', () => {
+    test('should turn a key value string into a record', () => {
+      const encoded = 'ONE=1|TWO=2';
+      const expected = new Map<string, string>([
+        ['ONE', '1'],
+        ['TWO', '2'],
+      ]);
+      const actual = keyValuesToMap(encoded);
+      expect(actual).toEqual(expected);
+    });
+
+    test('should turn a blank string into an empty record', () => {
+      const encoded = '';
+      const expected = new Map<string, string>();
+      const actual = keyValuesToMap(encoded);
+      expect(actual).toEqual(expected);
+    });
+
+    test('should omit a pair without a equal sign delimiter', () => {
+      const encoded = 'foo|ONE=1|bad';
+      const expected = new Map<string, string>([['ONE', '1']]);
+      const actual = keyValuesToMap(encoded);
+      expect(actual).toEqual(expected);
+    });
+
+    test('should omit empty pairs', () => {
+      const encoded = '||ONE=1|||||';
+      const expected = new Map<string, string>([['ONE', '1']]);
+      const actual = keyValuesToMap(encoded);
+      expect(actual).toEqual(expected);
+    });
+
+    test('should trim keys and values', () => {
+      const encoded = ' ONE = 1 | TWO      = 2         ';
+      const expected = new Map<string, string>([
+        ['ONE', '1'],
+        ['TWO', '2'],
+      ]);
+      const actual = keyValuesToMap(encoded);
+      expect(actual).toEqual(expected);
+    });
+  });
```

Comment:
I would like to add a test that covers something where the value includes an equal sign, maybe something like a connection string:
```
connectionString=Server=tcp:app-name.database.usgovcloudapi.net,1433;Initial Catalog=DATABASE_NAME;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;Authentication="Active Directory Default";|foo=true
```